pkg/controller/datavolume/util.go

```diff
@@ -44,85 +44,105 @@ const (
 )
 
 // renderPvcSpec creates a new PVC Spec based on either the dv.spec.pvc or dv.spec.storage section
-func renderPvcSpec(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume) (*v1.PersistentVolumeClaimSpec, error) {
+func renderPvcSpec(client client.Client, recorder record.EventRecorder, log logr.Logger, dv *cdiv1.DataVolume, pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaimSpec, error) {
```

Comment:
I think this would look cleaner if we do something like:

```go
	// If the PVC already exists, just return its spec
	if pvc != nil {
		return pvc.Spec.DeepCopy(), nil
	}
	if dv.Spec.PVC != nil {
		return dv.Spec.PVC.DeepCopy(), nil
	}
	if dv.Spec.Storage != nil {
		return pvcFromStorage(client, recorder, log, dv)
	}
``` 

`pvcFromStorage` would require fewer arguments too, wdyt?